test/edu/washington/escience/myria/operator/KeepAndSortOnMinTest.java

```diff
@@ -48,7 +48,7 @@ public void testKeepAndSortedOnMinValue() throws DbException {
     while (!keepmin.eos()) {
       keepmin.nextReady();
       long lastValue = -1;
-      List<TupleBatch> result = keepmin.getStreamingState().exportState();
+      List<TupleBatch> result = keepmin.getStreamingStates().get(0).exportState();
```

Comment:
Is the change to `getStreamingStates()` solely for the benefit of `Producer`? If so, and if none of the other `StreamingStateful` implementations can logically have more than one `StreamingState`, maybe we should include a convenience `getStreamingState()` method in the interface? Which callers actually expect a list from this method, instead of immediately calling `get(0)` on the result? It seems like every call site I see in this diff is made more awkward by this change.